video/out/hwdec/hwdec_vaapi_vk.c

```diff
@@ -41,6 +44,22 @@ static bool vaapi_vk_map(struct ra_hwdec_mapper *mapper)
         uint32_t size = p->desc.objects[id].size;
         uint32_t offset = p->desc.layers[n].offset[0];
 
+        // AMD drivers do not return the size in the surface description.
+        if (size == 0) {
+            size = lseek(fd, 0, SEEK_END);
+            if (size == -1) {
+                MP_ERR(mapper, "Cannot obtain size of object with fd %d: %s\n",
+                       fd, mp_strerror(errno));
+                return false;
+            }
+            off_t err = lseek(fd, 0, SEEK_SET);
+            if (err == -1) {
+                MP_ERR(mapper, "Failed to reset offset for fd %d: %s\n",
+                       fd, mp_strerror(errno));
+                return false;
+            }
+        }
```

Comment:
Would be slightly nicer to move this to a function. Also, is it required to reset the position? Even if, you could just fold both error checks into one.